### PR TITLE
Create Custom Schemas for Replicated.yml

### PR DESCRIPTION
**Pull request by christophermancini**
The scope of this PR is to create TS Interfaces that will be used to generate a JSONSchema file using the `typescript-json-schema` library. Left empty DocBlocks to be filled in later with descriptions.

Other small changes:
- Add indentation to tslint.json
- Create tsfmt.json
- Add command for tsfmt
- Add new command for tslint excluding test dir
- Run tsfmt
- Fix lint errors
- Add `typescript-json-schema` package to devDependencies
- Create .gitattributes file for normalizing line endings.

### CODE REVIEW

**Review by dexhorthy — CHANGES_REQUESTED** [review]
Hi @christophermancini -- thanks for putting this together, this looks like a great start on getting the libyaml definitions into Typescript. I had a few small comments:


1. I like the addition of `tsfmt`, but it looks like the formatting defaulted to 4 spaces instead of 2. Would it be possible to configure `tsfmt` so it keeps the whitespace consistent with the project conventions (2 spaces for indents)? Feel free to modify `tslint.json` or add a `tsfmt.json` if needed.
1. There are some codeclimate issues, but I think those can be ignored, its just picking up the TODOs from the libyaml snippets, which are probably still valid although it might be worth verifying if any of those have been deprecated for real so far.
1. Lot of commits, would you mind squashing them down to one?


All in all, looks good to go after the formatting thing is addressed.

**Comment by christophermancini** [comment]
@dexhorthy Thanks for the feedback! I expected to do a squash before the official merge, but I wanted to keep the commits separate to start in case there was something you didn't want to accept, e.g. the tsfmt run.

Yea, I noticed the codeclimate notices. I intended to remove the libyaml snippets once we finished all of the schemas and wrote tests to confirm they are working as expected.

I will adjust to 2 spaces for indents and look into configuring tsfmt to enforce it.

**Comment by dexhorthy** [comment]
Sounds like a plan :+1:

**Pull request merged**
